### PR TITLE
Modularize security groups

### PR DESCRIPTION
**Pull request by mpeamma**

### CODE REVIEW

File: modules/open_egress_sg/variables.tf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+variable "vpc_id" {
+  description = "VPC of Security Group"
+  type        = "string"
+}
+
+variable "tags" {
+  description = "Tags of Security Group"
+  type        = "map"
```
Comment (sean-nixon) [inline modules/open_egress_sg/variables.tf]: So this is confirmed to work? I know Terraform 0.11 has issues with passing lists as variables into modules but not sure if maps have similar problems.

File: modules/open_egress_sg/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+output "id" {
+  description = "Id of created security group"
+  value       = "${aws_security_group.sg.id}"
+}
+
+output "arn" {
+  description = "ARN of created security group"
+  value       = "${aws_security_group.sg.arn}"
+}
```
Comment (sean-nixon) [inline modules/open_egress_sg/outputs.tf]: Final newline plz

File: modules/open_egress_sg/main.tf [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+resource "aws_security_group_rule" "outbound" {
+  type              = "egress"
+  security_group_id = "${aws_security_group.sg.id}"
+  from_port         = "-1"
+  to_port           = "-1"
+  protocol          = "all"
+  cidr_blocks       = ["0.0.0.0/0"]
+}
```
Comment (sean-nixon) [inline modules/open_egress_sg/main.tf]: Final newline plz

File: apptier.tf [hunk 4]
```diff
@@ -136,36 +131,30 @@ resource "aws_security_group" "3tier-app-sg" {
...
 
-resource "aws_security_group_rule" "app-egress" {
-  type              = "egress"
-  security_group_id = "${aws_security_group.3tier-app-sg.id}"
-  from_port         = "-1"
-  to_port           = "-1"
-  protocol          = "all"
-  cidr_blocks       = ["0.0.0.0/0"]
-}
+
```
Comment (sean-nixon) [inline apptier.tf]: Remove extra newline

File: modules/open_egress_sg/variables.tf (hunk 1 above)
Comment (mpeamma) [reply modules/open_egress_sg/variables.tf]: Yep! Just double checked it

Comment (sean-nixon) [reply modules/open_egress_sg/variables.tf]: Sweet, that makes things easy

File: modules/open_egress_sg/outputs.tf (hunk 2 above)
Comment (mpeamma) [reply modules/open_egress_sg/outputs.tf]: Added

File: modules/open_egress_sg/main.tf (hunk 3 above)
Comment (mpeamma) [reply modules/open_egress_sg/main.tf]: Added

File: apptier.tf (hunk 4 above)
Comment (mpeamma) [reply apptier.tf]: Removed

**Pull request merged**